lib-rgbpanel/src/rgbpanel.cpp

```diff
@@ -55,27 +58,45 @@ RgbPanel::RgbPanel(uint32_t nColumns, uint32_t nRows, uint32_t nChain, RgbPanelT
 	m_ptColons = new struct TColon[m_nMaxPosition * m_nMaxLine];
 	assert(m_ptColons != nullptr);
 	SetColonsOff();
+
 }
 
 /**
  * Text
  */
 void RgbPanel::PutChar(char nChar, uint8_t nRed, uint8_t nGreen, uint8_t nBlue) {
-	if (__builtin_expect((static_cast<uint32_t>(nChar) >= cp437_font_size()), 0)) {
+
+	uint32_t font_size = 0;
+	uint32_t font_width = 0;
+	uint32_t font_height = 8;
+	uint32_t font_space = 0;
+
+	if (m_nFontID == 0){
+		font_size = cp437_font_size();
+		font_width = FONT_CP437_CHAR_W;
+	} else if (m_nFontID == 1) {
+		font_size = font_5x8_size();
+		font_width = FONT_5x8_CHAR_W;
+//		font_height = FONT_5x8_CHAR_H;
+		font_space = 1;		
+	}
+
+	if (__builtin_expect((static_cast<uint32_t>(nChar) >= font_size), 0)) {
 		nChar = ' ';
 	}
 
-	const auto nStartColumn = m_nPosition * FONT_CP437_CHAR_W;
-	auto nRow = m_nLine * m_nMaxPosition;
+	auto nStartColumn = m_nPosition * (font_width + font_space);
+	auto nRow = m_nLine * font_height; //* m_nMaxPosition;
 	const auto nColonIndex = m_nPosition + nRow;
 	const bool bShowColon = (m_ptColons[nColonIndex].nBits != 0);
 
-	for (uint32_t i = 0; i < FONT_CP437_CHAR_H; i++) {
-		uint32_t nWidth = 0;
+	uint32_t nColumn = 0;
```

Comment:
Why is nColumn taken out for-loop declaration?

Reply:
Whoops!